### ISSUE TITLE
Investigate `ModelTabletModeNoSuspend=1`

### ISSUE THREAD

**Issue by ChocolateLoverRaj** [WeirdTreeThing/cros-keyboard-map#19]
I'm not sure if this is from using keyd v2 vs v1, or if it's a redrix vs jinlon thing, but `ModelTabletModeNoSuspend=1` is now _messing up_ stuff.

With `ModelTabletModeNoSuspend=1`:
- Side volume buttons never get disabled
- Keyboard never gets disabled

Without `ModelTabletModeNoSuspend=1`:
- Side volume buttons never get disabled
- Keyboard gets disabled in tablet mode

Tested in NixOS. I'm not using this script, but maybe someone should investigate this option on another Chromebook and undo my [earlier pr](#4) if needed

**Comment by ChocolateLoverRaj** [comment]
Maybe we should use ectool to find out if the EC is detecting tablet mode properly. If it is, then maybe this can properly be fixed with changes to the Linux kernel.

**Comment by mussawarakram** [comment]
@crystalcommunication I'm experiencing a similar thing to your LAVA device with a PUJJO device - slightly different though in that my tablet switch is working as expected but my volume rocker does not (only batch propagates presses on resuming sleep, booting, and interestingly whenever the tablet switch changes state), however does nothing otherwise irrespective of the tablet switch state. 

Have there been any changes, workarounds or progress around that issue on LAVA?

**Comment by WeirdTreeThing** [comment]
I've honestly never looked into this issue.

**Comment by WeirdTreeThing** [comment]
On my morphius, the keyboard never turns off in tablet mode and the side volume keys always work.

**Comment by mussawarakram** [comment]
Fortunately for morphius there is a workaround script posted above which reassigns the keys to noop when the tablet mode switch activates - not ideal but usable.

I cannot wrap my head around what's going on with my pujjo though, and whatever is causing the side volume to register (in batches) only when specific events occur.

My suspicion is something is not quite right with the intel vbtn driver.

**Comment by WeirdTreeThing** [comment]
Yeah I don't like the whole noop thing. As for your issue, probably bad ec firmware? And IIRC the intel virtual button driver is just to make something in windows happy and not actually used, could be wrong there though.

**Comment by WeirdTreeThing** [comment]
Either way, the ec really should be responsible for disabling all this. I think chromeos does this in userspace though, because relying on the ec is unreliable. I've had devices where the keyboard gets disabled but the touchpad doesnt and vice versa.

**Comment by ChocolateLoverRaj** [comment]
So some Chromebooks have buggy EC firmware, but that firmware is also buggy on ChromeOS, not just MrChromebox firmware? And Jinlon just happens to be a nicely working one?

**Comment by WeirdTreeThing** [comment]
Well, MrChromebox ships custom EC firmware for most devices he supports. I don't think that would make any difference in this case, but it could.

**Comment by mussawarakram** [comment]
Just for anyone that jumps in and sees my issue above;

I've narrowed it down to an issue with the cros_ec drivers. I don't think it's the keyb one as that hasn't changed for a while but I noticed a regression in the 6.14 kernel which actually renders both the tablet switch and the volume button unusable. 

I was unable to pinpoint any module changes that might have caused it. 

From the 6.10 kernel to 6.13.x it works with the quirky behaviour I described above. 

6.9 and older kernels only seem to have the tablet switch working.

With ectool I could see all pressed being registered correctly and as expected.

**Comment by ChocolateLoverRaj** [comment]
So will it actually have to be every desktop environment's responsibility to disable the correct keys when in tablet mode?

**Comment by WeirdTreeThing** [comment]
I'm working on a keyd replacement specifically for chromebooks which will integrate automatic top row mapping. One of the features I plan to add is tablet mode handling to solve this problem. It will block all input events from the keyboard/trackpad when in tablet mode. The code is at https://github.com/WeirdTreeThing/croskbd.

**Comment by ChocolateLoverRaj** [comment]
> I did some digging and found that the ec is designed to never disable the keyboard when the system is powered on.

On my jinlon, as I was testing keyboard input in [my Rust OS](https://github.com/ChocolateLoverRaj/rust-os-tutorial)  (through PS/2), I found that interestingly when the Jinlon goes into tablet mode the keyboard keys do not create PS/2 Keyboard events but the side volume keys still continue to create keyboard events.

So I think this could also be Chromebook-specific.

**Comment by WeirdTreeThing** [comment]
Just tested on my pixelbook i just got, the keyboard doesn't send events in tablet mode, but the touchpad still does. The OS does choose to ignore the inputs from the touchpad.
